packages/xstate-store/src/setup.ts

```diff
@@ -0,0 +1,51 @@
+import { EventObject } from 'xstate';
+import { createStore } from './store';
+import {
+  Compute,
+  EventPayloadMap,
+  ExtractEventsFromPayloadMap,
+  Store,
+  StoreAssigner,
+  StoreContext,
+  StorePropertyAssigner
+} from './types';
+
+type MaybeZod<T> = T extends { _output: infer U } ? U : T;
```

Comment:
likely this should be OK with the other change:
```suggestion
type MaybeZod<T> = T extends { _output: infer U } ? U : never;
```

or once u fix constraints:
```suggestion
type MaybeZod<T extends { _output: unknown }> = T["_output"];
```